src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/TestManagerBuilder.cs

```diff
@@ -0,0 +1,149 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Extensions;
+using Microsoft.Azure.Commands.TestFx;
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.Rest;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Commands.Resources.Test.ScenarioTests
+{
+    public class TestManagerBuilder
```

Comment:
Can your emane this to something that implies that the class is intended to be a base class for all the resourceManager Tests?

Something like "ResourceManagerTestBase", TestManageVuilder sounds more like a builder class that should be packaged with TestManager.

Reply:
@praries880,
I didn't mean``` TestManagerBuilder``` to be a base class for all the ```Resources``` module tests, 
* I would day it's just a concise way to share a common test infrastructure setup among test classes. The idea was to hide the ```TestManager``` instance initialization in the base class constructor and share it via protected member. Otherwise I had to copy the same initialization sequence to each test class constructor. 
* If a customer needs a different infrastructure setup they can build a specific version of ```TestManager``` instance and inheritance would be redundant.